IntelliJ Rider fails to create settings pane

Version info:
OS: OSX 11.4
Rider 2021.1.3
Snyk Vulnerability Scanner 2.1.5
Snyk 1.623.0 (standalone)

**Steps:**

1. Authenticate to Snyk
2. Preferences > Tools > Snyk

**Result:** 

Loading spinner indefinitely displays.

**Expected result:**

Settings configuration pane is displayed.

**Stack trace:**

```
Cannot create configurable

com.intellij.diagnostic.PluginException: Cannot create class io.snyk.plugin.settings.SnykProjectSettingsConfigurable (classloader=PluginClassLoader(plugin=PluginDescriptor(name=Snyk Vulnerability Scanner, id=io.snyk.snyk-intellij-plugin, descriptorPath=plugin.xml, path=~/Library/Application Support/JetBrains/Rider2021.1/plugins/snyk-intellij-plugin, version=2.1.5, package=null), packagePrefix=null, instanceId=42, state=active))
	at com.intellij.serviceContainer.ComponentManagerImpl.instantiateClass(ComponentManagerImpl.kt:763)
	at com.intellij.serviceContainer.ComponentManagerImpl.instantiateClass(ComponentManagerImpl.kt:781)
	at com.intellij.openapi.options.ConfigurableEP$ClassProducer.createElement(ConfigurableEP.java:440)
	at com.intellij.openapi.options.ConfigurableEP.createConfigurable(ConfigurableEP.java:346)
	at com.intellij.openapi.options.ex.ConfigurableWrapper.createConfigurable(ConfigurableWrapper.java:42)
	at com.intellij.openapi.options.ex.ConfigurableWrapper.getConfigurable(ConfigurableWrapper.java:116)
	at com.intellij.openapi.options.ex.ConfigurableWrapper.cast(ConfigurableWrapper.java:91)
	at com.intellij.openapi.options.newEditor.SettingsTreeView.findConfigurableProject(SettingsTreeView.java:312)
	at com.intellij.openapi.options.newEditor.SettingsTreeView$MyRenderer.getTreeCellRendererComponent(SettingsTreeView.java:643)
	at java.desktop/javax.swing.plaf.basic.BasicTreeUI$NodeDimensionsHandler.getNodeDimensions(BasicTreeUI.java:3155)
	at java.desktop/javax.swing.tree.AbstractLayoutCache.getNodeDimensions(AbstractLayoutCache.java:493)
	at java.desktop/javax.swing.tree.VariableHeightLayoutCache$TreeStateNode.updatePreferredSize(VariableHeightLayoutCache.java:1344)
	at java.desktop/javax.swing.tree.VariableHeightLayoutCache$TreeStateNode.expand(VariableHeightLayoutCache.java:1465)
	at java.desktop/javax.swing.tree.VariableHeightLayoutCache$TreeStateNode.expand(VariableHeightLayoutCache.java:1272)
	at java.desktop/javax.swing.tree.VariableHeightLayoutCache.ensurePathIsExpanded(VariableHeightLayoutCache.java:967)
	at java.desktop/javax.swing.tree.VariableHeightLayoutCache.setExpandedState(VariableHeightLayoutCache.java:181)
	at java.desktop/javax.swing.plaf.basic.BasicTreeUI.updateExpandedDescendants(BasicTreeUI.java:1975)
	at java.desktop/javax.swing.plaf.basic.BasicTreeUI$Handler.treeExpanded(BasicTreeUI.java:4212)
	at java.desktop/javax.swing.JTree.fireTreeExpanded(JTree.java:2801)
	at java.desktop/javax.swing.JTree.setExpandedState(JTree.java:3702)
	at java.desktop/javax.swing.JTree.expandPath(JTree.java:2244)
	at java.desktop/javax.swing.plaf.basic.BasicTreeUI.toggleExpandState(BasicTreeUI.java:2616)
	at com.intellij.openapi.options.newEditor.SettingsTreeView$MyTreeUi.processMouseEvent(SettingsTreeView.java:836)
	at com.intellij.openapi.options.newEditor.SettingsTreeView$MyTree.processMouseEvent(SettingsTreeView.java:806)
	at java.desktop/java.awt.Component.processEvent(Component.java:6417)
	at java.desktop/java.awt.Container.processEvent(Container.java:2263)
	at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5027)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2321)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4859)
	at java.desktop/java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4918)
	at java.desktop/java.awt.LightweightDispatcher.processMouseEvent(Container.java:4547)
	at java.desktop/java.awt.LightweightDispatcher.dispatchEvent(Container.java:4488)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2307)
	at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2784)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4859)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:778)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:95)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:751)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:749)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:748)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:969)
	at com.intellij.ide.IdeEventQueue.dispatchMouseEvent(IdeEventQueue.java:906)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:836)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:449)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:797)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$9(IdeEventQueue.java:448)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:781)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:502)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:117)
	at java.desktop/java.awt.WaitDispatchSupport$2.run(WaitDispatchSupport.java:190)
	at java.desktop/java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:235)
	at java.desktop/java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:233)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.desktop/java.awt.WaitDispatchSupport.enter(WaitDispatchSupport.java:233)
	at java.desktop/java.awt.Dialog.show(Dialog.java:1063)
	at com.intellij.openapi.ui.impl.DialogWrapperPeerImpl$MyDialog.show(DialogWrapperPeerImpl.java:694)
	at com.intellij.openapi.ui.impl.DialogWrapperPeerImpl.show(DialogWrapperPeerImpl.java:436)
	at com.intellij.openapi.ui.DialogWrapper.doShow(DialogWrapper.java:1702)
	at com.intellij.openapi.ui.DialogWrapper.show(DialogWrapper.java:1661)
	at com.intellij.ide.actions.ShowSettingsUtilImpl.showSettingsDialog(ShowSettingsUtilImpl.java:90)
	at com.jetbrains.rider.settings.RiderShowSettingsUtilImpl.showSettingsDialog(RiderShowSettingsUtilImpl.kt:36)
	at com.intellij.ide.actions.ShowSettingsAction.perform(ShowSettingsAction.java:55)
	at com.intellij.ui.mac.MacOSApplicationProvider$Worker.lambda$initMacApplication$1(MacOSApplicationProvider.java:81)
	at com.intellij.ui.mac.MacOSApplicationProvider$Worker.lambda$submit$8(MacOSApplicationProvider.java:181)
	at com.intellij.openapi.application.TransactionGuardImpl.runWithWritingAllowed(TransactionGuardImpl.java:218)
	at com.intellij.openapi.application.TransactionGuardImpl.access$200(TransactionGuardImpl.java:21)
	at com.intellij.openapi.application.TransactionGuardImpl$2.run(TransactionGuardImpl.java:200)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:781)
	at com.intellij.openapi.application.impl.ApplicationImpl.lambda$invokeLater$4(ApplicationImpl.java:319)
	at com.intellij.openapi.application.impl.FlushQueue.doRun(FlushQueue.java:84)
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:133)
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:46)
	at com.intellij.openapi.application.impl.FlushQueue$FlushNow.run(FlushQueue.java:189)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:776)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:746)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:969)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:839)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:449)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:808)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$9(IdeEventQueue.java:448)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:781)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:496)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at com.intellij.serviceContainer.ComponentManagerImpl.instantiateClass(ComponentManagerImpl.kt:742)
	... 100 more
Caused by: java.lang.ClassCastException: class com.intellij.ui.components.BrowserLink cannot be cast to class com.intellij.ui.HyperlinkLabel (com.intellij.ui.components.BrowserLink and com.intellij.ui.HyperlinkLabel are in unnamed module of loader com.intellij.util.lang.PathClassLoader @668bc3d5)
	at io.snyk.plugin.ui.settings.ScanTypesPanel$$special$$inlined$panel$lambda$4.invoke(ScanTypesPanel.kt:102)
	at io.snyk.plugin.ui.settings.ScanTypesPanel$$special$$inlined$panel$lambda$4.invoke(ScanTypesPanel.kt:16)
	at com.intellij.ui.layout.RowBuilder$DefaultImpls.row(Row.kt:42)
	at com.intellij.ui.layout.Row.row(Row.kt:107)
	at com.intellij.ui.layout.LayoutBuilder.row(LayoutBuilder.kt)
	at com.intellij.ui.layout.RowBuilder$DefaultImpls.row$default(Row.kt:41)
	at io.snyk.plugin.ui.settings.ScanTypesPanel.<init>(ScanTypesPanel.kt:100)
	at io.snyk.plugin.ui.settings.ScanTypesPanel.<init>(ScanTypesPanel.kt:20)
	at io.snyk.plugin.ui.SnykSettingsDialog.<init>(SnykSettingsDialog.kt:50)
	at io.snyk.plugin.settings.SnykProjectSettingsConfigurable.<init>(SnykProjectSettingsConfigurable.kt:22)
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,9 @@
 # Snyk Vulnerability Scanner Changelog
 
+## [2.1.6]
+### Fixed
+- Fix [Exception when Settings panel invoked](https://github.com/snyk/snyk-intellij-plugin/issues/121) 
+
 ## [2.1.5]
 ### Changed
 - Check if Snyk Code (SAST) enabled for organisation on server side

--- a/gradle.properties
+++ b/gradle.properties
@@ -1,4 +1,4 @@
-pluginVersion=2.1.5
+pluginVersion=2.1.6
 pluginSinceBuild=202
 pluginUntilBuild=211.*
 #

--- a/src/main/kotlin/io/snyk/plugin/ui/settings/ScanTypesPanel.kt
+++ b/src/main/kotlin/io/snyk/plugin/ui/settings/ScanTypesPanel.kt
@@ -25,7 +25,9 @@ class ScanTypesPanel(
     private var snykCodeCheckbox: JBCheckBox? = null
     private var snykCodeQualityCheckbox: JBCheckBox? = null
     private var snykCodeComment: JLabel? = null
-    private var snykCodeAlertLinkLabel: HyperlinkLabel? = null
+    private var snykCodeAlertLinkLabel = HyperlinkLabel().apply {
+        this.isVisible = false
+    }
 
     val panel = panel {
         row {
@@ -98,10 +100,8 @@ class ScanTypesPanel(
             }
         }
         row {
-            snykCodeAlertLinkLabel = browserLink("", "")
+            snykCodeAlertLinkLabel()
                 .withLargeLeftGap()
-                .component as HyperlinkLabel
-            snykCodeAlertLinkLabel?.isVisible = false
         }
     }
 
@@ -147,9 +147,10 @@ class ScanTypesPanel(
 
     fun showSnykCodeAlert(message: String, linkText: String = "", url: String = "", forceShow: Boolean = false) {
         if (settings.sastOnServerEnabled || forceShow) {
-            snykCodeAlertLinkLabel?.isVisible = message.isNotEmpty()
-            snykCodeAlertLinkLabel?.setHyperlinkText(message, linkText, "")
-            snykCodeAlertLinkLabel?.setHyperlinkTarget(url)
+            snykCodeAlertLinkLabel.isVisible = message.isNotEmpty()
+            // todo: change to setTextWithHyperlink() after move to sinceId >= 211
+            snykCodeAlertLinkLabel.setHyperlinkText(message, linkText, "")
+            snykCodeAlertLinkLabel.setHyperlinkTarget(url)
         }
     }
 

